### DISCUSSION TITLE
Lost diagram - Looking for recovery

### DISCUSSION THREAD

**Discussion by LucasVati** [hediet/vscode-drawio#404] [Q&A]
Hello,

I have been working on a graph for hours in the draw.io vscode extension. I was saving the file often.
When finished the work, I exported to png, the resolution was too low, so I tried to convert to draw.io.png, which resulted in the same png as before.
So I deleted the draw.io.png file and wanted to come back to the draw.io file to find out other ways to export the graph.

However, the draw.io file is not there. I think a draw.io copy of the diagram hasn't been kept when converting to draw.io.png.
Am I right ? If yes, I am very upset. There should be a copy kept in case this happens !
Anyway, is there a way to recover the draw.io file ? Maybe in some hidden extension memory ?

**Comment by hediet** [comment]
> So I deleted the draw.io.png file and

Can you look in your paper bin? The diagram meta data is stored inside the png file.

This extension has "export" and "convert". Export creates a copy (which is safe to delete), convert modifies the diagram (which shouldn't be deleted).
You might also have luck with VS Codes local timeline feature.

**Reply by LucasVati** [reply]
Thank you for your answer. I checked in the paper bin, it wasn't inside.
Finally, I searched for "drawio" in my entire computer, and just found a recovery in the following path : home\lucas\.vscode-server\data\User\History\1e70733a. Very nice ! 

I wish I knew that the 'drawio.png' file shouldn't be deleted. 
When converting, it could be nice to ask the user if he/she wants to keep a copy of the original .drawio file, or to specify that the original .drawio will be replaced ? I imagine I am not the first one that faced this adventure, so it could help future users.

Have a nice week-end !
